tests/main/views/test_frameworks.py

```diff
@@ -378,6 +395,152 @@ def test_update_framework_catches_db_errors(self, db_commit, open_example_framew
+        self.framework.status = framework_start_status
+        db.session.add(self.framework)
+        db.session.commit()
+
+        self.client.post(
+            '/frameworks/g-cloud-10',
+            data=json.dumps({
+                'frameworks': {'status': framework_new_status},
+                'updated_by': 'example user'
+            }),
+            content_type="application/json"
+        )
+
+        self._assert_declaration_matches(expected_declaration_data)
+
+    def test_updating_framework_to_pending_uses_first_contact_information_entry(self):
+        """Digital Marketplace theoretically supports multiple contact informations per supplier. When copying contact
+        information across to fill in the supplier declaration, we want to take the details of the first (lowest id)
+        contact information associated with the account."""
+        db.session.add(ContactInformation(supplier_id=1,
+                                          contact_name='Our awesome contact',
+                                          email='2_is_the_new_1@contact.com',
+                                          address1="Twilight's Library",
+                                          city="Ponyville",
+                                          postcode="PO42"))
+        db.session.commit()
+
+        self.client.post(
+            '/frameworks/g-cloud-10',
+            data=json.dumps({
+                'frameworks': {'status': "pending"},
+                'updated_by': 'example user'
+            }),
+            content_type="application/json"
+        )
+
+        self._assert_declaration_matches(self.EXPECTED_FRAMEWORK_DECLARATION_SUPPLIER_INFO)
+
+    def test_updating_framework_includes_keys_with_none_value_for_null_supplier_account_information(self):
+        supplier = Supplier.query.filter(Supplier.supplier_id == 1).first()
+        supplier.registration_date = None
+        supplier.vat_number = None
+        supplier.registered_name = None
+
+        contact_info = ContactInformation.query.filter(ContactInformation.supplier_id == supplier.supplier_id).first()
+        contact_info.city = None
+
+        db.session.add(supplier)
+        db.session.add(contact_info)
+        db.session.commit()
+
+        self.client.post(
+            '/frameworks/g-cloud-10',
+            data=json.dumps({
+                'frameworks': {'status': "pending"},
+                'updated_by': 'example user'
+            }),
+            content_type="application/json"
+        )
+
+        expected_declaration_data = self.EXPECTED_FRAMEWORK_DECLARATION_SUPPLIER_INFO.copy()
+        expected_declaration_data[KEY_REGISTRATION_DATE] = None
+        expected_declaration_data[KEY_VAT_NUMBER] = None
+        expected_declaration_data[KEY_REGISTERED_NAME] = None
+        expected_declaration_data[KEY_REGISTRATION_TOWN] = None
+
+        self._assert_declaration_matches(expected_declaration_data)
+
```

Comment:
It might be nice to have one further test, where we see a  declaration with pre-existing values for some of the "magic" fields have these values overwritten when the framework status changes to pending again, and all the other "non-magic" fields remain unchanged.

There's no test here with existing declaration data - it would definitely be good to have at least one test where we check that we don't inadvertently wipe all the existing declaration data when we do the copy over.

Reply:
Done. 👍